internal/stores/s3.go

```diff
@@ -153,7 +153,7 @@ func (s *S3Store) Upload(ctx context.Context, objects ...string) error {
 		objPathrelativeToSourceRepo := filepath.Join(pwd, o)
 		metadataPath := fmt.Sprintf("%s.%s", objPathrelativeToSourceRepo, s.Options.MetaDataFileExtension)
 		logger.V(2).Infof("writing metadata to %s", metadataPath)
-		if err := os.WriteFile(metadataPath, blob, 0644); err != nil {
+		if err := afero.WriteFile(s.fsys, metadataPath, blob, 0644); err != nil {
```

Comment:
I noticed the `metadataPath` var is equal to the `fname` path in GCS, can we name these similarly or create a helper func to just return the `metdataPath` consistently across stores.